packages/cli/src/services/translationIO.ts

```diff
@@ -0,0 +1,284 @@
+import fs from "fs"
+import { dirname } from "path"
+import PO from "pofile"
+import https from "https"
+import glob from "glob"
+import { format as formatDate } from "date-fns"
+
+const getCreateHeaders = (language) => ({
+  "POT-Creation-Date": formatDate(new Date(), "yyyy-MM-dd HH:mmxxxx"),
+  "MIME-Version": "1.0",
+  "Content-Type": "text/plain; charset=utf-8",
+  "Content-Transfer-Encoding": "8bit",
+  "X-Generator": "@lingui/cli",
+  Language: language,
+})
+
+// Main sync method, call "Init" or "Sync" depending on the project context
+export default function syncProcess(config, options) {
+  const successCallback = (project) => {
+    console.log(`\n----------\nProject successfully synchronized. Please use this URL to translate: ${project.url}\n----------`)
+  }
+
+  const failCallback = (errors) => {
+    console.error(`\n----------\nSynchronization with Translation.io failed: ${errors.join(', ')}\n----------`)
+  }
+
+  init(config, options, successCallback, (errors) => {
+    if (errors.length && errors[0] === 'This project has already been initialized.') {
+      sync(config, options, successCallback, failCallback)
+    }
+    else {
+      failCallback(errors)
+    }
+  })
+}
+
+// Initialize project with source and existing translations (only first time!)
+// Cf. https://translation.io/docs/create-library#initialization
+function init(config, options, successCallback, failCallback) {
+  const sourceLocale  = config.sourceLocale || 'en'
+  const targetLocales = config.locales.filter((value) => value != sourceLocale)
+  const paths         = poPathsPerLocale(config)
+
+  let segments = {}
+
+  targetLocales.forEach((targetLocale) => {
+    segments[targetLocale] = []
+  })
+
+  // Create segments from source locale PO items
+  paths[sourceLocale].forEach((path) => {
+    let raw = fs.readFileSync(path).toString()
+    let po = PO.parse(raw)
+
+    po.items.filter((item) => !item.obsolete).forEach((item) => {
+      targetLocales.forEach((targetLocale) => {
+        let newSegment = createSegmentFromPoItem(item)
+
+        segments[targetLocale].push(newSegment)
+      })
+    })
+  })
+
+  // Add translations to segments from target locale PO items
+  targetLocales.forEach((targetLocale) => {
+    paths[targetLocale].forEach((path) => {
+      let raw = fs.readFileSync(path).toString()
+      let po = PO.parse(raw)
+
+      po.items.filter((item) => !item.obsolete).forEach((item, index) => {
+        segments[targetLocale][index].target = item.msgstr[0]
+      })
+    })
+  })
+
+  let request = {
+    "client": "lingui",
+    "version": require("../package.json").version,
```

Comment:
Everything looks good to me, except this `require`.  What does try to require? Lingui version?

Reply:
Also are TypeScript complains a bit on the Ci, could you take a look?

Reply:
> Everything looks good to me, except this `require`. What does try to require? Lingui version?

Yes, the API requires the client version. It's useful to debug potential client-server errors more easily. We also have a process to trigger a warning if the client version become way too old and create issues.

Maybe is there a better way to get the version?

> Also are TypeScript complains a bit on the Ci, could you take a look?

I'm not sure how I could find it on the CI, the only thing I see in `lint-pr-title` is "Error: No release type found in pull request title "Cloud service providers & Translation.io"".

But I remember seeing some warnings locally, I'll try to fix them tomorrow.

Thanks again for the review!

Reply:
https://github.com/lingui/js-lingui/runs/3172321494 Here it is `tsc` failing

Reply:
You can replace to get the correct version to this:
```
"version": require('@lingui/core/package.json').version
```

It will require the @lingui/core package and get the current version of the workspace, much safer than using relative paths :)